bus/functools_test.go

```diff
@@ -0,0 +1,69 @@
+package bus
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+)
+
+func TestFunctionHelloWorld(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	res := ""
+	e := NewEndpoints(consumer, publisher)
+	f, err := e.Function("helloworld", func(arg interface{}) error {
+		res = fmt.Sprintf("%v", arg)
+		wg.Done()
+		return nil
+	})
+	if err != nil {
+		t.Errorf("cannot create function, %v", err)
+	}
+
+	value := "Hello world"
+	err = f.MustSucceed(value)
+	wg.Wait()
```

Comment:
Of course this sync on a waitgroup is only for this test and works well in this setting with only one function, but I think we should explicitly make clear that this is not a pattern that will work in a distributed setting.